source/System.cpp

```diff
@@ -315,6 +316,9 @@ void System::Load(const DataNode &node, Set<Planet> &planets)
 				object.message = &UNINHABITEDPLANET;
 		}
 	}
+	// Print a warning if this system has a position that is the galactic origin.
```

Comment:
What would you suggest as an alternative? Before this PR, the standard was to check if the system had a name. This PR ensures everything to be serialized is named. Position is the simplest - 0, 0 is not valid in Endless Sky due to neighboring system proximity. I'm not sure what other limitations we can come up with for a System's invariant other than "name and position" -- I certainly agree that a system with no stellar objects is something we should allow to exist.

For anything modifiable by a GameEvent, we can't truly say whether the class object was loaded fully, because any load will set `isDefined`, even if it's just a change node that e.g. updates the fleets.

Reply:
The best I can think of is to have a boolean variable that gets initialized with false and that gets set to true when a position is set for a system. (You still track if the position was ever set, but do allow 0,0 as valid position, as long as it is explicitly set.)